v2/pkg/genruntime/conditions/ready_condition_builder.go

```diff
@@ -6,6 +6,9 @@ Licensed under the MIT license.
 package conditions
 
 const (
+	// Auth reasons
+	ReasonSubscriptionMismatch = "SubscriptionMismatch"
```

Comment:
minor: IMO this fits OK into "other reasons", it doesn't need its own section.

Reply:
we'll have to add more reasons like `CredentialSecretNotFound` into it when we do per-rg secrets, so thought lets have a separate place for these.

Reply:
Ah, good point. Resolving this as I agree with you above.